src/Disks/ObjectStorages/MetadataStorageFromPlainObjectStorage.cpp

```diff
@@ -161,6 +169,26 @@ std::optional<StoredObjects> MetadataStorageFromPlainObjectStorage::getStorageOb
     return std::nullopt;
 }
 
+MetadataStorageFromPlainObjectStorage::ObjectMetadataEntryPtr
+MetadataStorageFromPlainObjectStorage::getObjectMetadataEntryWithCache(const std::string & path) const
+{
+    auto get = [&] -> ObjectMetadataEntryPtr
+    {
+        auto object_key = object_storage->generateObjectKeyForPath(path, std::nullopt /* key_prefix */);
+        if (auto metadata = object_storage->tryGetObjectMetadata(object_key.serialize()))
+            return std::make_shared<ObjectMetadataEntry>(metadata->size_bytes, metadata->last_modified.epochTime());
+        return nullptr;
```

Comment:
If we once failed to get object metadata, for example, because s3 was temporarily unavailable, then looks like we will add nullptr to cache for this path and it would always be nullptr and never updated, should we instead check the error code because of which we failed to get metadata, and if it is key_doesnt_exist, then probably ok to leave nullptr in cache, but for another reason seems better not to cache it?

Reply:
Fixed so null pointer is not cached.
I think storing a nullptr for non-existed key would be wrong in the present implementation, since I don't update the cache when a file is written.